### PR TITLE
Add CORS to REST endpoint

### LINKED ISSUE #708: CORS not enabled
**Bug Description**
When querying /api/v0/info for the block0 hash the server throws a CORS not enabled.

**Steps to Reproduce**
1. write a small JS script that constructs an HTTP request for data to /api/v0/info
2. ensure you've enabled CORS in the above HTTP request
3. execute it in the browser targetting the jormungandr REST (from a different origin)

**Expected behavior**
Jormungandr REST server returns the data

**Actual behaviour**
Jormungandr REST server throws a CORS not enabled error

### PR DESCRIPTION
**Pull request by CodeSandwich**
Fixes #708 

Tested with https://www.w3schools.com/code/tryit.asp?filename=G77D6NQHMTGV, master branch refuses to work and prints errors to console, this one works fine if configured.

### CODE REVIEW

File: doc/configuration/network.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ p2p:
 
 - *listen*: listen address
 - *pkcs12*: certificate file (optional)
+- *cors*: (optional) CORS configuration, if not provided, CORS is disabled
+  - *allowed_origins*: (optional) allowed origins, if none provided, echos request origin
+  - *max_age_secs*: (optional) maximum CORS caching time in secons, if none provided, caching is disabled
```
Comment (NicolasDP) [inline doc/configuration/network.md]:
```suggestion
  - *max_age_secs*: (optional) maximum CORS caching time in seconds, if none provided, caching is disabled
```

File: doc/quickstart/02_passive_node.md [hunk 2]
```diff
@@ -63,6 +63,9 @@ Description of the fields:
 - `rest`: (optional) Configuration of the REST endpoint.
     - `listen`: _address_:_port_ to listen for requests
     - `pkcs12`: (optional) Certificate file
+    - `cors`: (optional) CORS configuration, if not provided, CORS is disabled
+      - `allowed_origins`: (optional) allowed origins, if none provided, echos request origin
+      - `max_age_secs`: (optional) maximum CORS caching time in secons, if none provided, caching is disabled
```
Comment (NicolasDP) [inline doc/quickstart/02_passive_node.md]:
```suggestion
      - `max_age_secs`: (optional) maximum CORS caching time in seconds, if none provided, caching is disabled
```

**Comment by CodeSandwich** [comment]
Thanks, I've amended your fixes :+1:

File: jormungandr/src/settings/start/config.rs [hunk 3]
```diff
@@ -40,6 +40,17 @@ pub struct ConfigLogSettings {
...
+    pub cors: Option<Cors>,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Cors {
+    /// If none provided, echos request origin
+    #[serde(default)]
+    pub allowed_origins: Vec<String>,
+    /// If none provided, CORS responses won't be cached
+    pub max_age_secs: Option<usize>,
```
Comment (mzabaluev) [inline jormungandr/src/settings/start/config.rs]: Please avoid using `usize` for integers that don't relate to size of objects in memory. This should be `u64`.

Comment (NicolasDP) [reply jormungandr/src/settings/start/config.rs]: Good point actually. We have a type for that in `jormungandr-lib::time`

Comment (CodeSandwich) [reply jormungandr/src/settings/start/config.rs]: Oops, I've oversighted it. I'll fix it in next PR.

**Pull request merged**